### PR TITLE
Use new Input Selection

### LINKED ISSUE #1641: [Bug report] Sending to a newly minted NFT or Alias has side effects
## Bug description

Briefly describe the bug.
Sending to a newly minted NFT or Alias has side effects.
- Transaction and subsequent transactions returns n-1 transaction instead of the transaction that was just created.
- NFT or Alias is included in the transaction for no apparent reason.

## Rust version

Which version of Rust are you running?

- Rust version: 

## Version

Which version of the library are you using?

- Version number, commit, or branch: `2.0.3-rc.8`

## Hardware specification

What hardware are you using?

- Operating system:
- RAM:
- Cores:
- Device:

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

1. Mint NFT or Alias
2. Send to Alias or NFT from the same wallet
3. Make subsequent Transactions

## Expected behaviour

Describe what you expect to happen.

- Transactions created in wallet.rs return that transaction not the previous one.
- Transactions created in wallet.rs do not contain inputs and outputs for the sake of it without making changes.

## Actual behaviour

Describe what actually happens.
- Transaction and subsequent transactions returns n-1 transaction instead of the transaction that was just created.
- NFT or Alias is included in the transaction for no apparent reason.
- This persists until destroying and recreating the account manager

Example
Transaction 1 - Create Alias:

- Transaction: https://explorer.shimmer.network/testnet/transaction/0x2b22a17006c2500d6725fbda269a6705f65d51c4b1389bbc0190b6e3032151ad
- Returned: Correct transaction with id: `0x2b22a17006c2500d6725fbda269a6705f65d51c4b1389bbc0190b6e3032151ad`

Transaction 2 - Send to Alias

- Transaction: https://explorer.shimmer.network/testnet/transaction/0xb751d64f5db265565ac9dcd43d6f9bb3b8d4175f1087febe8e19f39261930c14
- Returned: Previous Transaction with id: `0x2b22a17006c2500d6725fbda269a6705f65d51c4b1389bbc0190b6e3032151ad`
- Includes my Alias for no reason?

Transaction 3 - Send to Mark’s Alias

- Transaction: https://explorer.shimmer.network/testnet/transaction/0xa75bf0324847f9e2c58a7308a32842eeb19ce5e31bae3e5e8d206bfa7ac2fd3f
- Returned: Previous Transaction with id: `0xb751d64f5db265565ac9dcd43d6f9bb3b8d4175f1087febe8e19f39261930c14`
- Includes my Alias for no reason?

## Errors

Paste any errors that you see, including logs, errors, or screenshots.

N/A

### LINKED ISSUE #1662: Mint NFT fails if NFT present in the account
## Bug description

NotEnoughBalance error on minting NFT if the NFT is present in the account, was either:
- minted to self
- sent from another account

## Rust version

- Rust version: rustc 1.65.0

## Version

- Version number, commit, or branch: branch=develop#8870f4c78171a3a5bacc7486258a90fab8ab5985

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

1. clone https://github.com/wedvjin/iota-balance-nft
2. ```cargo run```
3.  can also be reproduced by creating a new fresh account address, and by sending an nft from another account, then trying to mint from the fresh (funded) account

## Expected behaviour

Considering that the balance is sufficient, mint shall succeed.

## Actual behaviour

Error NotEnoughBalance occurs.

## Errors

The account balance is:
```
AccountBalance { base_coin: BaseCoinBalance { total: 499707700, available: 499575400 }, required_storage_deposit: RequiredStorageDeposit { alias: 0, basic: 127800, foundry: 0, nft: 132300 }, native_tokens: [], nfts: [NftId(0x8a5b30a25f97b51a9d62d0736bab9361b32300dff1a022845a30eab12dab15c1), NftId(0xff8464c8b8904dbb0494bc9fd9fcc1980d8ae47dae6beb563588d4fe3a9e67b6)], aliases: [], foundries: [], potentially_locked_outputs: {} }
```

But the error is:
```
Client(NotEnoughBalance { found: 52300, required: 104600 })
```

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account/operations/transaction/input_selection.rs [hunk 1]
```diff
@@ -44,32 +43,43 @@ impl AccountHandle {
         );
 
         let current_time = self.client.get_time_checked().await?;
-        let bech32_hrp = self.client.get_bech32_hrp().await?;
+
+        // Filter inputs to not include inputs that require additional outputs for storage deposit return or could be
+        // still locked TODO should maybe be done in ISA directly ?
```
Comment (Thoralf-M) [inline src/account/operations/transaction/input_selection.rs]: Outputs that are still locked should be skipped in the ISA, is that not happening yet?

Comment (thibault-martinez) [reply src/account/operations/transaction/input_selection.rs]: They are now fed to forbidden outputs

File: src/account/operations/transaction/options.rs [hunk 2]
```diff
@@ -59,8 +61,7 @@ pub struct TransactionOptionsDto {
     pub custom_inputs: Option<Vec<OutputId>>,
     #[serde(rename = "mandatoryInputs", default)]
     pub mandatory_inputs: Option<Vec<OutputId>>,
-    #[serde(rename = "allowBurning", default)]
-    pub allow_burning: bool,
+    pub burn: Option<BurnDto>,
```
Comment (Thoralf-M) [inline src/account/operations/transaction/options.rs:64]: Bindings need to be adapted for this change (and it should be in the changelog)

Comment (thibault-martinez) [reply src/account/operations/transaction/options.rs:64]: 😡😡😡

**Review by Thoralf-M — DISMISSED** [review]
Changelog still not updated for the burn rename :trollface:
But can be done in another PR

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**